src/popup/searchinterface.js

```diff
@@ -33,15 +33,7 @@ function SearchInterface(popup) {
 function view(ctl, params) {
     var self = this;
 
-    let url = "";
-    const authUrl = helpers.parseAuthUrl(window?.location?.href ?? null);
-    if (this.popup.settings.authRequested && authUrl) {
-        url = new BrowserpassURL(authUrl);
-    } else {
-        url = new BrowserpassURL(this.popup.settings.origin);
-    }
-
-    const host = url.hostname;
+    const host = helpers.getCurrentUrl(this.popup.settings).hostname;
```

Comment:
```suggestion
    const url = helpers.getCurrentUrl(this.popup.settings);
    const host = url.isValid ? url.hostname : "";
```

(I feel like it's both safe and more correct to align this with your new code)